extensions/rv32im/circuit/src/hintstore/mod.rs

```diff
@@ -334,211 +342,135 @@ impl<F: PrimeField32> InstructionExecutor<F> for Rv32HintStoreChip<F> {
-        };
+        let mem_ptr_msl = mem_ptr >> ((RV32_REGISTER_NUM_LIMBS - 1) * RV32_CELL_BITS);
+        let num_words_msl = num_words >> ((RV32_REGISTER_NUM_LIMBS - 1) * RV32_CELL_BITS);
+        // TODO(ayush): see if this can be moved to fill_trace_row
+        self.bitwise_lookup_chip.request_range(
+            mem_ptr_msl << (RV32_REGISTER_NUM_LIMBS * RV32_CELL_BITS - self.pointer_max_bits),
+            num_words_msl << (RV32_REGISTER_NUM_LIMBS * RV32_CELL_BITS - self.pointer_max_bits),
+        );
+
+        for word_index in 0..(num_words as usize) {
+            let offset = *trace_offset + word_index * width;
+            let mut row: &mut Rv32HintStoreCols<F> = trace[offset..offset + width].borrow_mut();
 
-        for word_index in 0..num_words {
             if word_index != 0 {
-                memory.increment_timestamp();
-                memory.increment_timestamp();
+                row.is_buffer = F::ONE;
+                row.from_state.timestamp = F::from_canonical_u32(state.memory.timestamp);
+
+                state.memory.increment_timestamp();
+                state.memory.increment_timestamp();
             }
 
-            let data: [F; RV32_REGISTER_NUM_LIMBS] =
+            // TODO(ayush): why is hint_stream F if composed of u8 limbs?
+            let data_f: [F; RV32_REGISTER_NUM_LIMBS] =
                 std::array::from_fn(|_| streams.hint_stream.pop_front().unwrap());
-            // let (write, _) = memory.write(
-            //     e,
-            //     F::from_canonical_u32(mem_ptr + (RV32_REGISTER_NUM_LIMBS as u32 * word_index)),
-            //     &tmp_convert_to_u8s(data),
-            // );
-            // record.hints.push((data, write));
-        }
-
-        self.height += record.hints.len();
-        self.records.push(record);
-
-        let next_state = ExecutionState {
-            pc: from_state.pc + DEFAULT_PC_STEP,
-            timestamp: memory.timestamp(),
-        };
-        Ok(next_state)
-    }
-
-    fn get_opcode_name(&self, opcode: usize) -> String {
-        if opcode == HINT_STOREW.global_opcode().as_usize() {
-            String::from("HINT_STOREW")
-        } else if opcode == HINT_BUFFER.global_opcode().as_usize() {
-            String::from("HINT_BUFFER")
-        } else {
-            unreachable!("unsupported opcode: {}", opcode)
-        }
-    }
-}
+            let data: [u8; RV32_REGISTER_NUM_LIMBS] =
+                data_f.map(|byte| byte.as_canonical_u32() as u8);
 
-impl<F: Field> ChipUsageGetter for Rv32HintStoreChip<F> {
-    fn air_name(&self) -> String {
-        "Rv32HintStoreAir".to_string()
-    }
-
-    fn current_trace_height(&self) -> usize {
-        self.height
-    }
-
-    fn trace_width(&self) -> usize {
-        Rv32HintStoreCols::<F>::width()
-    }
-}
+            for half in 0..(RV32_REGISTER_NUM_LIMBS / 2) {
+                self.bitwise_lookup_chip
+                    .request_range(data[2 * half] as u32, data[2 * half + 1] as u32);
```

Comment:
if easy, move this to `generate_proof_input`

Reply:
done